### ISSUE TITLE
[FIX] obsidian vault is too big with all kinds of media. image and pdf exclude not enough

### ISSUE THREAD

**Issue by huachuman** [khoj-ai/khoj#913]
## Describe the bug
I can't use khoj because my vault contains too many different kinds of files. not just media but .js, and all kinds of things from plugins that I don't need khoj to index either.

- Client:
    - [x] Obsidian
    - [x] Windows

## Additional context
It might be best to use "include only .md files" or give users a whitelist instead of allowing to remove only images and pdfs...

**Comment by debanjum** [comment]
Hey @huachuman, are you seeing non-markdown files being synced from you Obsidian vault after you've updated your Khoj Obsidian plugin to index only markdown files? The Khoj Obsidian plugin only syncs Markdown, PDF and Image files by default, not any other files types. So if this is happening it'd be a bug.

Steps to ensure that only markdown files from your Obsidian vault are being synced with Khoj:

1. **Only sync Markdown files from the Khoj Obsidian Plugin**: Update the Sync settings on your Khoj Obsidian plugin settings page to look like the image I shared below:
   <img width="420" alt="Settings to only sync Markdown files from the Khoj Obsidian Plugin" src="https://github.com/user-attachments/assets/0b4d7249-eabe-478e-ad04-943ad1dd0a7e">
2. **Delete all previously synced files**: Click "Manage" and than "Delete All" on the Files card on the [web settings page](https://app.khoj.dev/settings)
3. **Manual Force Sync**: Click the *Update* button to Force Sync your Vault from the Khoj Obsidian plugin settings page
4. **Check the currently synced files on the web app** again to ensure that only markdown files from your Obsidian vault are synced

**Comment by huachuman** [comment]
I keep getting these errors in console when i try

index.html:1 Access to fetch at 'https://app.khoj.dev/api/content?client=obsidian' from origin 'app://obsidian.md' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
app.khoj.dev/api/content?client=obsidian:1     Failed to load resource: net::ERR_FAILED
plugin:khoj:1089        Uncaught (in promise) TypeError: Failed to fetch
    at updateContentIndex (plugin:khoj:1089:28)
    at async eval (plugin:khoj:2723:34)

**Comment by debanjum** [comment]
Oh, that's strange. Which version of the Khoj Obsidian plugin are you on? (It should be visible under Settings > Community Plugins > > Installed Plugins section).

I am able to Force sync to `https://app.khoj.dev` using the latest Khoj Obsidian plugin (version **1.23.2**)

**Comment by huachuman** [comment]
Thanks. I tried to update the plugin and am on 1.23.2 (was previously on 1.23.1) but still getting this:

Access to fetch at 'https://app.khoj.dev/api/content?client=obsidian' from origin 'app://obsidian.md' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
plugin:khoj:1089     PUT https://app.khoj.dev/api/content?client=obsidian net::ERR_FAILED 504
updateContentIndex @ plugin:khoj:1089
plugin:khoj:1089        Uncaught (in promise) TypeError: Failed to fetch
    at updateContentIndex (plugin:khoj:1089:28)
    at async eval (plugin:khoj:1430:39)
updateContentIndex @ plugin:khoj:1089

**Comment by debanjum** [comment]
Hi @huachuman , are you still seeing the CORS issue on trying to index your Obsidian vault with Khoj? If so, can you open a separate issue to track that and maybe share a screenshot of your Khoj obsidian plugin settings page. I'm unable to reproduce the CORS indexing issue on my end.

The original issue to configure the Khoj Obsidian plugin to only index markdown files was resolved a few months ago. So I'll close this issue to reduce confusion and separate it from the CORS issue you're facing.
